app/Http/Controllers/RankingController.php

```diff
@@ -20,27 +20,53 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Beatmap;
+use App\Models\UserStatistics;
+use App\Transformers\UserStatisticsTransformer;
+use Request;
+
 class RankingController extends Controller
 {
-    protected $section = 'ranking';
+    protected $section = 'rankings';
 
-    public function getOverall()
-    {
-        return view('ranking.overall');
-    }
+    const PAGE_SIZE = 20;
+    const MAX_RESULTS = 10000;
 
-    public function getCountry()
+    public function index($mode, $type, $page = 1)
     {
-        return view('ranking.country');
-    }
+        $max_pages = ceil($this::MAX_RESULTS / $this::PAGE_SIZE);
 
-    public function getCharts()
-    {
-        return view('ranking.charts');
-    }
+        if (!array_key_exists($mode, Beatmap::MODES)) {
+            abort(404);
+        }
 
-    public function getMapper()
-    {
-        return view('ranking.mapper');
+        $stats = UserStatistics\Model::getClass($mode)
+            ->with('user')
+            ->whereHas('user', function ($userQuery) {
+                $userQuery->default();
+            })
+            ->orderBy('rank_score', 'desc')
+            ->limit($this::PAGE_SIZE);
+
+        $page = clamp(get_int($page), 1, $max_pages) - 1;
```

Comment:
set together with `maxPages` so `->offset` can be called together with thing above?